Kernel.cc

```diff
@@ -1367,3 +1370,277 @@ short Kernel::findIndexNode( char * path , IndexNode& inode )
 	return indexNodeNumber ;
 }
 
+int Kernel::link(char *oldpath, char *newpath) {
+  	// get the full path
+	char *fullPath = getFullPath(oldpath);
```

Comment:
This works exactly like create, by traversing the directory. There's probably a better way to do it